File: app/src/main/java/io/shelfy/architecture/data/source/remote/mappers/MovieVideoMapper.java
package io.shelfy.architecture.data.source.remote.mappers;

import androidx.annotation.NonNull;

import io.shelfy.architecture.ApiConstants;
import io.shelfy.architecture.data.source.remote.response.MovieVideoJson;
import io.shelfy.architecture.common.Mapper;
import io.shelfy.architecture.domain.entity.MovieVideo;

public class MovieVideoMapper implements Mapper<MovieVideoJson, MovieVideo> {
    @NonNull
    @Override
    public MovieVideo map(@NonNull MovieVideoJson movieVideoJson) {
        return new MovieVideo(ApiConstants.YOUTUBE_BASE_URL + movieVideoJson.getKey());
    }
}

File: app/src/main/java/io/shelfy/architecture/data/mappers/MovieVideoMapper.kt
package io.shelfy.architecture.data.mappers

import com.vladk.architecture.api.ApiConstants
import com.vladk.architecture.api.response.MovieVideoJson
import com.vladk.architecture.domain.model.MovieVideo
import com.vladk.architecture.domain.mappers.Mapper

class MovieVideoMapper : Mapper<MovieVideoJson, MovieVideo> {

    override fun map(from: MovieVideoJson): MovieVideo {
        return with(from) {
            MovieVideo(videoUrl = ApiConstants.YOUTUBE_BASE_URL + key)
        }
    }
}
